src/Bedrock.Framework/Protocols/ConsumableArrayBufferWriter.cs

```diff
@@ -132,6 +128,12 @@ public void Consume(int count)
             {
                 _index = 0;
                 _consumedCount = 0;
+                if (Capacity >= DefaultInitialBufferSize * 2)
```

Comment:
I think we want to return the buffer on consumed regardless of the size no?

Reply:
If it's the Default size why return it? The moment anyone wrote anything we would ask for it back, and if they've finished with it they should dispose...